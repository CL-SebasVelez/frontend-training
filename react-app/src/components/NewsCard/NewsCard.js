import React from 'react';
import PropTypes from 'prop-types';

function NewsCard(props) {
  const handleClick = (url) => {
    window.open(url, '_blank');
  };
  const open = props.url ? 'c-pointer' : '';
  return (
    <div
      className={`card ${open} mb-20`}
      onClick={() => (props.url ? handleClick(props.url) : void 0)}
    >
      <div className="card-image">
        <img src={props.image} alt="" />
      </div>
      <div className="card-title mt-5">
        <p>{props.title}</p>
      </div>
      <div className="card-content">
        <p>{props.description}</p>
      </div>
    </div>
  );
}

NewsCard.propTypes = {
  url: PropTypes.string,
  image: PropTypes.string,
  title: PropTypes.string.isRequired,
  description: PropTypes.string,
};

NewsCard.defaultProps = {
  image:
    'https://www.bnd.com/latest-news/ppc7fl/picture222958020/alternates/LANDSCAPE_768/NEWSnew.jpg',
  title: 'News',
  description: '',
};

export default NewsCard;
