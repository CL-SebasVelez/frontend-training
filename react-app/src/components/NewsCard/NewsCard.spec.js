import React from 'react';
import renderer from 'react-test-renderer';

import NewsCard from './NewsCard';

describe('NewsCard component', () => {
  it('should render with default props', () => {
    const tree = renderer.create(<NewsCard />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should launch error when pass wrong prop types', () => {
    const { url, image, title, description } = 10;
    expect(() =>
      shallow(
        <NewsCard
          url={url}
          image={image}
          title={title}
          description={description}
        />
      )
    ).toThrowError();
  });

  it('should render all props as passed it', () => {
    const title = 'New NewsCard';
    const image =
      'https://www.bnd.com/latest-news/ppc7fl/picture222958020/alternates/LANDSCAPE_768/NEWSnew.jpg';
    const url =
      'https://www.bnd.com/latest-news/ppc7fl/picture222958020/alternates/LANDSCAPE_768/NEWSnew.jpg';
    const description = 'Description NewsCard';

    const testRenderer = renderer.create(
      <NewsCard
        title={title}
        image={image}
        url={url}
        description={description}
      />
    );
    const testInstance = testRenderer.root;
    expect(testInstance.findByType(NewsCard).props.title).toBe(title);
    expect(testInstance.findByType(NewsCard).props.image).toBe(image);
    expect(testInstance.findByType(NewsCard).props.url).toBe(url);
    expect(testInstance.findByType(NewsCard).props.description).toBe(
      description
    );
  });
});
