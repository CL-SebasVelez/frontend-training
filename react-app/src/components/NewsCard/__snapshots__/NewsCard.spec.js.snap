// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsCard component should render with default props 1`] = `
<div
  className="card  mb-20"
  onClick={[Function]}
>
  <div
    className="card-image"
  >
    <img
      alt=""
      src="https://www.bnd.com/latest-news/ppc7fl/picture222958020/alternates/LANDSCAPE_768/NEWSnew.jpg"
    />
  </div>
  <div
    className="card-title mt-5"
  >
    <p>
      News
    </p>
  </div>
  <div
    className="card-content"
  >
    <p>
      
    </p>
  </div>
</div>
`;

exports[`NewsCard component should render without default props 1`] = `
<div
  className="card  mb-20"
  onClick={[Function]}
>
  <div
    className="card-image"
  >
    <img
      alt=""
      src="https://www.bnd.com/latest-news/ppc7fl/picture222958020/alternates/LANDSCAPE_768/NEWSnew.jpg"
    />
  </div>
  <div
    className="card-title mt-5"
  >
    <p>
      News
    </p>
  </div>
  <div
    className="card-content"
  >
    <p>
      
    </p>
  </div>
</div>
`;
